src/controller/AutoCommissioner.cpp

```diff
@@ -363,9 +363,9 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
     {
         completionStatus.failedStage = MakeOptional(report.stageCompleted);
         ChipLogError(Controller, "Failed to perform commissioning step %d", static_cast<int>(report.stageCompleted));
-        if (report.stageCompleted == CommissioningStage::kAttestationVerification)
+        if (report.Is<AdditionalErrorInfo>())
         {
-            if (report.Is<AdditionalErrorInfo>())
+            if (report.stageCompleted == CommissioningStage::kAttestationVerification)
```

Comment:
I know you didn't add this, but you can just remove this - it's redundant.

Reply:
I think it's fine as-is - the report is a variant, which jives with having these separate. It's not a huge deal either way - this report is just passed to the commissioning delegate, and the variant is a dynamic allocation, so it's not like either way is terrible. If you want to keep them separate, you might want to re-name AdditionalErrorInfo into something more specific. I named it when it was the only error info being passed back.